core/util_test.go

```diff
@@ -54,116 +69,234 @@ type DrandTest2 struct {
+func getSleepDuration() time.Duration {
+	if os.Getenv("CIRCLE_CI") != "" {
+		fmt.Println("--- Sleeping on CIRCLECI")
+		return time.Duration(600) * time.Millisecond
+	}
+	return time.Duration(500) * time.Millisecond
 }
 
 // NewDrandTest creates a drand test scenario with initial n nodes and ready to
 // run a DKG for the given threshold that will then launch the beacon with the
 // specified period
-func NewDrandTest2(t *testing.T, n, thr int, period time.Duration) *DrandTest2 {
-	dt := new(DrandTest2)
-	drands, _, dir, certPaths := BatchNewDrand(n, false,
-		WithCallOption(grpc.WaitForReady(true)),
+func NewDrandTestScenario(t *testing.T, n, thr int, period time.Duration) *DrandTestScenario {
+	dt := new(DrandTestScenario)
+
+	drands, _, dir, certPaths := BatchNewDrand(
+		t, n, false, WithCallOption(grpc.WaitForReady(true)),
 	)
+
 	dt.t = t
 	dt.dir = dir
 	dt.certPaths = certPaths
+	dt.groupPath = path.Join(dt.dir, "group.toml")
 	dt.n = n
 	dt.thr = thr
 	dt.period = period
 	dt.clock = clock.NewFakeClock()
-	dt.nodes = make([]*Node, 0, n)
-	for i, drand := range drands {
-		node := dt.newNode(drand, dt.certPaths[i])
+	dt.nodes = make([]*MockNode, 0, n)
+
+	for i, drandInstance := range drands {
+		node := newNode(dt.clock.Now(), dt.certPaths[i], drandInstance)
 		dt.nodes = append(dt.nodes, node)
 	}
-	dt.groupPath = path.Join(dt.dir, "group.toml")
+
 	return dt
 }
 
 // Ids returns the list of the first n ids given the newGroup parameter (either
 // in the original group or the reshared one)
-func (d *DrandTest2) Ids(n int, newGroup bool) []string {
+// Deprecated: Rename this to addresses to align naming
+func (d *DrandTestScenario) Ids(n int, newGroup bool) []string {
 	nodes := d.nodes
 	if newGroup {
 		nodes = d.resharedNodes
 	}
-	var ids []string
+
+	var addresses []string
 	for _, node := range nodes[:n] {
-		ids = append(ids, node.addr)
+		addresses = append(addresses, node.addr)
 	}
-	return ids
+
+	return addresses
 }
 
 // RunDKG runs the DKG with the initial nodes created during NewDrandTest
-func (d *DrandTest2) RunDKG() *key.Group {
+func (d *DrandTestScenario) RunDKG() *key.Group {
 	// common secret between participants
 	secret := "thisisdkg"
-	root := d.nodes[0]
-	controlClient, err := net.NewControlClient(root.drand.opts.controlPort)
+
+	leaderNode := d.nodes[0]
+	controlClient, err := net.NewControlClient(leaderNode.drand.opts.controlPort)
 	require.NoError(d.t, err)
-	// the root node will return the group over this channel
+
+	d.t.Log("[RunDKG] Start")
+
+	// the leaderNode will return the group over this channel
 	var wg sync.WaitGroup
 	wg.Add(d.n)
+
 	// first run the leader and then run the other nodes
 	go func() {
-		gp, err := controlClient.InitDKGLeader(d.n, d.thr, d.period, d.catchupPeriod, testDkgTimeout, nil, secret, testBeaconOffset)
+		d.t.Log("[RunDKG] Leader init")
+
+		// TODO: Control Client needs every single parameter, not a protobuf type. This means that it will be difficult to extend
```

Comment:
Well actually there is the control client struct which is only a wrapper around the protobuf - we could use directly here the protobuf and send it via the generic client.